### ISSUE TITLE
[fragment-matcher] Error of missing directives when using `federation: true` 

### ISSUE THREAD

**Issue by mikestaub** [dotansimha/graphql-code-generator#2406]
**Describe the bug**
I cannot generate the fragments when using federation.

**To Reproduce**
https://codesandbox.io/embed/graphql-codegen-issue-template-z5t3q

**Expected behavior**
Should run without errors.

**Comment by dotansimha** [comment]
@mikestaub why do you pass `federation: true` to `fragment-matcher`? It's not a valid option there.

Federation is a backend feature, and you should use it with `typescript-resolvers` only, and use it in your Federation backend.

**Comment by dotansimha** [comment]
@kamilkisiela and @ardatan noticed a different issue, some types are skipped and not being generated by `fragment-matcher`.

**Comment by ardatan** [comment]
@mikestaub `fragment-matcher` generates introspection data only for unions and interfaces. You don't have any of those.

**Comment by mikestaub** [comment]
@dotansimha I am passing `federation: true` to `fragment-matcher` because if I don't I get this error: `Unknown directive "key".` 

As a workaround, I am simply including this file in the `schema` list field:

```
scalar _Any
scalar _FieldSet

# a union of all types that use the @key directive
union _Entity = Asset | Product | User | UserGroup

type _Service {
  sdl: String
}

extend type Query {
  _entities(representations: [_Any!]!): [_Entity]!
  _service: _Service!
}

directive @external on FIELD_DEFINITION
directive @requires(fields: _FieldSet!) on FIELD_DEFINITION
directive @provides(fields: _FieldSet!) on FIELD_DEFINITION
directive @key(fields: _FieldSet!) on OBJECT | INTERFACE
```

As defined here: https://www.apollographql.com/docs/apollo-server/federation/federation-spec/


Do you have an example of the `fragment-matcher` working with a federated schema?

**Comment by dotansimha** [comment]
Thanks @mikestaub . You are right. 
At the moment, you can add those manually (add another item for `schema` field) and it will workaround this issue for your. 
Me and @ardatan are thinking about doing a core change in the codegen that might fix that.

**Comment by kamilkisiela** [comment]
#2508 should handle that

**Comment by ardatan** [comment]
Is this issue still relevant?

**Comment by dotansimha** [comment]
Fixed in `v1.10.0`.
